simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/follow_polyline_trajectory_action.cpp

```diff
@@ -0,0 +1,584 @@
+         condition of "Is remaining time to front waypoint nan?"
+      */
+      if (
+        std::isnan(remaining_time_to_front_waypoint) or
+        isDefinitelyLessThan(
+          remaining_time_to_front_waypoint,
+          remaining_time_to_arrival_to_front_waypoint + step_time)) {
+        if (std::rotate(
+              std::begin(parameter->shape.vertices), std::begin(parameter->shape.vertices) + 1,
+              std::end(parameter->shape.vertices));
+            not parameter->closed) {
+          parameter->shape.vertices.pop_back();
+        }
+        /*
+           The OpenSCENARIO standard does not define the behavior when the
+           value of Timing.domainAbsoluteRelative is "relative". The standard
+           only states "Definition of time value context as either absolute or
+           relative", and it is completely unclear when the relative time
+           starts.
+
+           This implementation has interpreted the specification as follows:
+           Relative time starts from the start of FollowTrajectoryAction or
+           from the time of reaching the previous "waypoint with arrival time".
+        */
+        if (
+          not parameter->timing_is_absolute and not std::isnan(remaining_time_to_front_waypoint)) {
+          relative_timing_offset = entity_status.time;
+        }
+        return tick();  // tail recursion
+      } else {
+        throw common::SimulationError(
+          "Vehicle ", std::quoted(entity_status.name), " arrived at the waypoint in trajectory ",
+          remaining_time_to_front_waypoint,
+          " seconds earlier than the specified time. This may be due to unrealistic conditions of "
+          "arrival time specification compared to vehicle parameters and dynamic constraints.");
+      }
+    }
+
+    const auto desired_velocity = [&]() {
+      /*
+         Note: The followingMode in OpenSCENARIO is passed as variable
+         dynamic_constraints_ignorable. the value of the variable is
+         `followingMode == position`.
+      */
+      if (parameter->dynamic_constraints_ignorable) {
+        return normalize(target_position - position) * desired_speed;  // [m/s]
+      } else {
+        /*
+           Note: The vector returned if dynamic_constraints_ignorable == true
+           ignores parameters such as the maximum rudder angle of the vehicle
+           entry. In this clause, such parameters must be respected and the
+           rotation angle difference of the z-axis center of the vector must be
+           kept below a certain value.
+        */
+        throw common::SimulationError("The followingMode is only supported for position.");
+      }
+    }();
+
+    /*
+       Note: If obstacle avoidance is to be implemented, the steering behavior
+       known by the name "collision avoidance" should be synthesized here into
+       steering.
+    */
+    const auto steering = desired_velocity - velocity;
```

Comment:
Longitudinal planning algorithm was recommended to implement inside [LongitudinalSpeedPlanner class](https://github.com/tier4/scenario_simulator_v2/blob/master/simulation/traffic_simulator/src/behavior/longitudinal_speed_planning.cpp)

Reply:
I think [LongitudinalSpeedPlanner class](https://github.com/tier4/scenario_simulator_v2/blob/master/simulation/traffic_simulator/src/behavior/longitudinal_speed_planning.cpp) should be cleaned up in the near future, so at this time please implement here.